### PR TITLE
 Added attribute to validate Umbraco form route strings (ufprt) 

### PR DESCRIPTION
**Pull request by ronaldbarendse**
This is an alternative to PR https://github.com/umbraco/Umbraco-CMS/pull/5891, but differs on the following:
- Returns an exception with details why the validation failed (similar to `ValidateAntiForgeryTokenAttribute`)
- Also validates the MVC area

The last commit also removes the `UmbracoAntiForgeryAdditionalDataProvider`, as it doesn't add additional data for direct routes and can thus only optionally validate the `ufprt` token.

Maybe the attribute can include some additional documentation why and where it can be used, something like: https://github.com/umbraco/Umbraco-CMS/blob/14acb1421da2ee641e866dca96eb6e92259cfa1c/src/Umbraco.Web/Mvc/UmbracoProxyRouteOnlyAttribute.cs#L7

### CODE REVIEW

**Comment by Shazwazza** [comment]
Have merged in, will cherry pick to the 8.1 branch and make ammends from there.

File: src/Umbraco.Web/Mvc/HttpUmbracoFormRouteStringException.cs [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        /// </summary>
+        public HttpUmbracoFormRouteStringException()
+        { }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="HttpUmbracoFormRouteStringException" /> class.
+        /// </summary>
+        /// <param name="info">The <see cref="T:System.Runtime.Serialization.SerializationInfo" /> that holds the serialized object data about the exception being thrown.</param>
+        /// <param name="context">The <see cref="T:System.Runtime.Serialization.StreamingContext" /> that holds the contextual information about the source or destination.</param>
+        private HttpUmbracoFormRouteStringException(SerializationInfo info, StreamingContext context)
```
[the remark this answers is not in the document]
Comment (ronaldbarendse) [reply src/Umbraco.Web/Mvc/HttpUmbracoFormRouteStringException.cs]: This just makes sure the exception can be serialized/deserialized correctly and is following the Microsoft guidelines for designing custom exceptions, see: https://docs.microsoft.com/en-us/previous-versions/dotnet/netframework-4.0/ms229064(v=vs.100) and https://stackoverflow.com/questions/94488/what-is-the-correct-way-to-make-a-custom-net-exception-serializable.

[the remark this answers is not in the document]
Comment (Shazwazza) [reply src/Umbraco.Web/Mvc/HttpUmbracoFormRouteStringException.cs]: Okidoke will add back in, not seen a need for that before but must exist in some cases - should prob look at our other custom exceptions.

[the remark this answers is not in the document]
Comment (ronaldbarendse) [reply src/Umbraco.Web/Mvc/HttpUmbracoFormRouteStringException.cs]: This constructor is basically a requirement for all classes that implement `ISerializable` (besides adding `SerializableAttribute`), like `System.Exception` does: https://docs.microsoft.com/en-us/dotnet/api/system.runtime.serialization.iserializable?redirectedfrom=MSDN&view=netframework-4.8.

**Pull request merged**